### PR TITLE
Fix MM-40428

### PR DESCRIPTION
**Pull request by hahmadia**
#### Summary

Fixes MM-40428

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-40428

### CODE REVIEW

File: server/app/blocks.go [hunk 1]
```diff
@@ -124,6 +126,18 @@ func (a *App) DeleteBlock(c store.Container, blockID string, modifiedBy string)
...
 	}
 
+	if block.Type == model.TypeImage {
+		fileName, fileIDExists := block.Fields["fileId"]
+		if fileName, fileIDIsString := fileName.(string); fileIDExists && fileIDIsString {
+			filePath := filepath.Join(block.WorkspaceID, block.RootID, fileName)
+			err = a.filesBackend.RemoveFile(filePath)
+
+			if err != nil {
+				return err
```
Comment (wiggin77) [inline server/app/blocks.go:136]: I think we need to broadcast the block delete even if the image delete fails.

Comment (hahmadia) [reply server/app/blocks.go]: Agreed. Good call. I've decided to just log the error and proceed.

File: server/app/blocks.go [hunk 2]
```diff
@@ -124,6 +126,18 @@ func (a *App) DeleteBlock(c store.Container, blockID string, modifiedBy string)
...
 	}
 
+	if block.Type == model.TypeImage {
+		fileName, fileIDExists := block.Fields["fileId"]
+		if fileName, fileIDIsString := fileName.(string); fileIDExists && fileIDIsString {
+			filePath := filepath.Join(block.WorkspaceID, block.RootID, fileName)
+			err = a.filesBackend.RemoveFile(filePath)
+
+			if err != nil {
+				a.logger.Error("Error deleting image file", mlog.Err(err))
```
Comment (wiggin77) [inline server/app/blocks.go:136]: I recommend logging the filepath so administrators know which file they need to manually clean up.

**Review by wiggin77 — APPROVED** [review]
LGTM :+1:

**Review by ogi-m — APPROVED** [review]
LGTM!

**Pull request merged**